spec/requests/internal/classified_listings_spec.rb

```diff
@@ -0,0 +1,20 @@
+require "rails_helper"
+
+RSpec.describe "/internal/listings", type: :request do
+  describe "PUT /internal/listings/:id" do
+    let(:admin) { create(:user, :super_admin) }
+    let(:classified_listing) { create(:classified_listing, user_id: admin.id) }
+
+    before do
+      sign_in admin
+    end
+
+    it "clears listing cache" do
+      perform_enqueued_jobs do
+        put "/internal/listings/#{classified_listing.id}", params: {
```

Comment:
This doesn't actually check if the cache was cleared. It just checks that the update went without raising exceptions.
I would check that the job was enqueued.